TLM/TLM/UI/MainMenu/MainMenuPanel.cs

```diff
@@ -160,61 +142,90 @@ public class SizeProfile {
         public void OnUpdate(GlobalConfig config) {
             UpdatePosition(new Vector2(config.Main.MainMenuX, config.Main.MainMenuY));
 
-            if (started) {
-                DetermineProfile(config);
+            if (isStarted_) {
                 UpdateAllSizes();
                 Invalidate();
             }
         }
 
-        private void DetermineProfile(GlobalConfig conf) {
-            int profileIndex = conf.Main.TinyMainMenu ? 0 : 1;
-            activeProfile = SIZE_PROFILES[profileIndex];
-        }
-
         public void UpdateAllSizes() {
             UpdateSize();
             UpdateDragSize();
             UpdateButtons();
         }
 
         private void UpdateSize() {
-            width = activeProfile.MENU_WIDTH;
-            height = activeProfile.MENU_HEIGHT;
+            float buttonSize = ScaledSize.GetButtonSize();
+            // 6 buttons horizontal row, 2 rows + titlebar
+            width = ScaledSize.GetWidth();
+            height = ScaledSize.GetHeight();
+        }
+
+        /// <summary>Calculates button and panel sizes based on the screen resolution.</summary>
+        internal class ScaledSize {
+            internal const int NUM_ROWS = 2;
+            internal const int NUM_COLS = 6;
+
+            /// <summary>Calculate width of main menu panel, based on button width and spacings.</summary>
+            /// <returns>Width of the panel.</returns>
+            internal static float GetWidth() {
+                // 6 buttons + spacings (each 1/8 of a button)
+                const float ALL_SPACINGS = NUM_COLS * 0.125f;
+                return GetButtonSize() * (6f + ALL_SPACINGS);
+            }
+
+            internal static float GetHeight() {
+                // 2 button rows + spacing (1/8th) + titlebar
+                return (GetButtonSize() * 2.125f) + GetTitlebarHeight();
+            }
```

Comment:
I think you forgot to add spacings(right and bottom) because it looks like this (ModTools debugView):

![image](https://user-images.githubusercontent.com/19638970/74887863-79a30b80-537c-11ea-8591-a24c5ac8c387.png)

Reply:
Added missing spacing right and bottom